pymeasure/instruments/kepco/kepcobop.py

```diff
@@ -0,0 +1,180 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+# List of Kepco BOP's (Vmax,Imax) [single channel output]
```

Comment:
Do I understand it correctly, that it has channels? If so, you might consider using a `Channel` class.

Reply:
Hi Benedikt, thanks for all the comments, they are very helpful.

Yes, there do exist some models which have two output channels. However, it turns out they are not very sophisticated in design and are essentially two independent power supplies in a single unit. They each have their own digital interface card and GPIB address, though they share a single, physical connector. The implementation for each sub-power supply looks to be the same as for the single channel output models (I have had a brief scan of the manual), since they use the same interface card (BIT 4886) as the single channel models.

What do you think is the best route to go down? I could imagine creating a class for the two channel model, e.g. `Kepco2X504` (two channels of 50 V, 4 A) that is simply two `Kepco504` as each channel, though I'm not sure what the advantage would be over just declaring two single channel outputs since they behave that anyway. There doesn't seem to be any functionality of the unit as a whole that affects each channel simultaneously.

Reply:
If they have different GPIB addresses (very strange setup), they need two different instruments.
Therefore, you do not need to create a `Channel`.